src/main/kotlin/Node.kt

```diff
@@ -72,6 +73,8 @@ sealed class Node(
     }
     abstract fun receive(message: Message): Node
 
+    //TODO add receiving a list would avoid having to do the convoluted calls transforming a list in a typedArray and then using the * to destruct the array?
+    // as in node.add(*heartbeats.map { SentMessage(it, network.clock) }.toTypedArray())
```

Comment:
Yes correct, we just can't use `vararg` which I don't think we really use anyway

Reply:
I'm going to refactor, we will have a overload one method receives a single object and another a list, it will be more ergonomic that way, those array conversions are a bit hard to read

Reply:
yep agreed